desktopexporter/internal/app/types/api-types.ts

```diff
@@ -1,11 +1,15 @@
+export type RootSpan = {
+  serviceName: string;
+  name: string;
+  startTime: string;
+  endTime: string;
+};
+
 export type TraceSummary = {
+  traceID: string;
   hasRootSpan: boolean;
-  rootServiceName: string;
-  rootName: string;
-  rootStartTime: string;
-  rootEndTime: string;
+  rootSpan: RootSpan;
```

Comment:
`hasRootSpan` indicates that `rootSpan` is optional, but the type doesn't. Would be:

```typescript
rootSpan?: RootSpan;
```
or
```typescript
rootSpan: RootSpan | null
```

We also likely don't need `hasRootSpan` and can use the presence / lack of presence of `rootSpan` for that signal.

Reply:
Yeah, I was debating that. This is in part what I meant by “I don’t like how my front end is organized”. I’m going to go change it.